satpy/writers/cf_writer.py

```diff
@@ -220,11 +395,12 @@ def _collect_datasets(self, datasets, kwargs):
             except KeyError:
                 new_datasets = [ds.copy(deep=True)]
             for new_ds in new_datasets:
-                start_times.append(new_ds.attrs.pop("start_time", None))
-                end_times.append(new_ds.attrs.pop("end_time", None))
+                start_times.append(new_ds.attrs.get("start_time", None))
+                end_times.append(new_ds.attrs.get("end_time", None))
                 datas[new_ds.attrs['name']] = self.da2cf(new_ds,
-                                                         kwargs.get('epoch',
-                                                                    EPOCH))
+                                                         epoch=kwargs.get('epoch', EPOCH),
+                                                         flatten_attrs=kwargs.get('flatten_attrs', False),
+                                                         exclude_attrs=kwargs.get('exclude_attrs', None))
```

Comment:
In the spirit of limiting the `kwargs` usage to a minimum, could these be proper optional arguments in the `save_dataset` prototype, or is it creating problems ?